sqliterepo/replication.c

```diff
@@ -638,11 +638,18 @@ sqlx_transaction_end(struct sqlx_repctx_s *ctx, GError *err)
 	EXTRA_ASSERT(ctx->sq3->db != NULL);
 
 	if (err) {
+		if (g_strrstr(err->message, "SQLITE_NOTADB") != NULL
+				|| g_strrstr(err->message, "SQLITE_CORRUPT") != NULL) {
+			sq3->corrupted = TRUE;
+		}
 		ctx->any_change = 0;
 		rc = sqlx_exec(ctx->sq3->db, "ROLLBACK");
 		if (rc != SQLITE_OK && rc != SQLITE_DONE) {
 			GRID_WARN("ROLLBACK failed! (%d/%s) %s", rc,
 					sqlite_strerror(rc), sqlite3_errmsg(ctx->sq3->db));
+			if (rc == SQLITE_NOTADB || rc == SQLITE_CORRUPT) {
```

Comment:
if rollback was successful, the DB may still seen as corrupted due to l.643, is it intentional ?

Reply:
Yes, it's intentional. If an error indicates that the database is corrupt, it's best to rebuild it in any case.